service/controller/resource/tccp/create.go

```diff
@@ -319,9 +322,38 @@ func (r *Resource) newParamsMain(ctx context.Context, cr infrastructurev1alpha2.
 }
 
 func (r *Resource) newParamsMainInstance(ctx context.Context, cr infrastructurev1alpha2.AWSCluster, t time.Time) (*template.ParamsMainInstance, error) {
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
-		return nil, microerror.Mask(err)
+	var err error
+	// We need to fetch the ControlPlane CR for once because it needs to be passed to hamaster interface to get the mapopings.
+	var cp infrastructurev1alpha2.AWSControlPlane
+	{
+		var list infrastructurev1alpha2.AWSControlPlaneList
+
+		err := r.k8sClient.CtrlClient().List(
+			ctx,
+			&list,
+			client.InNamespace(cr.GetNamespace()),
+			client.MatchingLabels{label.Cluster: key.ClusterID(&cr)},
+		)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if len(list.Items) == 0 {
+			return nil, microerror.Mask(notFoundError)
+		}
+		if len(list.Items) > 1 {
+			return nil, microerror.Mask(tooManyCRsError)
+		}
+
+		cp = list.Items[0]
+	}
```

Comment:
That is not necessary. We can just pass the `cr` because what is important is the cluster label and all CRs have that. 

```suggestion
```